### PR TITLE
Add eq benchmark for StringArray/StringViewArray

### LINKED ISSUE #5903: Implement benchmarks for `compare_op` for `GenericBinaryView`
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
Part of https://github.com/apache/arrow-rs/issues/5374

@XiangpengHao  added very nice support to compare kernels for GenericBinaryArray in https://github.com/apache/arrow-rs/pull/5900

**Describe the solution you'd like**
We should add benchmark coverage for these kernels to ensure we don't regress their performance and to help with future optimizations

**Describe alternatives you've considered**
Add tests for `=` and `!=` in https://github.com/apache/arrow-rs/blob/master/arrow/benches/comparison_kernels.rs

**Additional context**

### PR DESCRIPTION
**Pull request by XiangpengHao**
# Which issue does this PR close?

Closes #5903.

# Rationale for this change
 
Now we have a rough idea of the performance between `StringViewArray` and `StringArray`.

tldr: for this workload, the `StringViewArray` is 3x faster than `StringArray` 
```
Eq/StringArray          time:   [17.044 ms 17.077 ms 17.115 ms]
                        change: [-3.1537% -2.8287% -2.4900%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 5 outliers among 100 measurements (5.00%)
  3 (3.00%) high mild
  2 (2.00%) high severe

Eq/StringViewArray      time:   [6.6760 ms 6.7019 ms 6.7263 ms]
                        change: [-2.4557% -1.9821% -1.5484%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 17 outliers among 100 measurements (17.00%)
  4 (4.00%) low severe
  13 (13.00%) low mild

Neq/StringArray         time:   [16.842 ms 16.865 ms 16.895 ms]
                        change: [-3.1768% -2.7754% -2.3814%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 5 outliers among 100 measurements (5.00%)
  2 (2.00%) high mild
  3 (3.00%) high severe

Neq/StringViewArray     time:   [6.8212 ms 6.8465 ms 6.8770 ms]
                        change: [-2.0508% -1.5512% -1.0159%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 6 outliers among 100 measurements (6.00%)
  3 (3.00%) high mild
  3 (3.00%) high severe
```

======

I do notice the mileage varies a lot with different workloads, and I spent quite some time to understand why.
Maybe it's worthwhile to present a more comprehensive evaluation/understanding of when/where to use StringViewArray. But I guess that is after we finish the DataFusion integration.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
Thank you @XiangpengHao  --  can you please put these benchmarks with the existing comparison benchmarks? 


https://github.com/apache/arrow-rs/blob/master/arrow/benches/comparison_kernels.rs

Maybe it would be a good time to add like/ilike benchmarks for StringView too

**Comment by XiangpengHao** [comment]
> Maybe it would be a good time to add like/ilike benchmarks for StringView too

like/ilike will be supported after #5931, I'll add them once it is merged.

File: arrow/benches/comparison_kernels.rs [hunk 1]
```diff
@@ -138,6 +150,45 @@ fn add_benchmark(c: &mut Criterion) {
         b.iter(|| eq(&arr_month_day_nano_b, &scalar).unwrap())
     });
 
+    let mut rng = seedable_rng();
+    let mut array_gen = make_string_array(1024 * 1024 * 8, &mut rng);
+    let string_left = StringArray::from_iter(array_gen);
+    let string_view_left = StringViewArray::from_iter(string_left.iter());
+
+    // reference to the same rng to make sure we generate **different** array data,
```
Comment (alamb) [inline arrow/benches/comparison_kernels.rs:158]: 👍

**Review by alamb — APPROVED** [review]
LGTM -- thanks @XiangpengHao  -- I don't think there is any reason to do a special neq benchmark upon further reflection as the underlying code is the same

**Comment by alamb** [comment]
Thanks @XiangpengHao !

**Comment by alamb** [comment]
> > Maybe it would be a good time to add like/ilike benchmarks for StringView too
> 
> like/ilike will be supported after #5931 is merged, I'll add them once it is merged.

FYI PR to add thm in https://github.com/apache/arrow-rs/pull/5936
